hack/generator/pkg/astmodel/storage_conversion_factories.go

```diff
@@ -0,0 +1,360 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package astmodel
+
+import (
+	"fmt"
+	"github.com/Azure/k8s-infra/hack/generator/pkg/astbuilder"
+	"github.com/dave/dst"
+	"github.com/pkg/errors"
+	"go/token"
+)
+
+// createPropertyConversion tries to create a property conversion between the two provided properties, using all of the
+// available conversion functions in priority order to do so.
+func createPropertyConversion(
+	sourceProperty *PropertyDefinition,
+	destinationProperty *PropertyDefinition,
+	knownLocals map[string]struct{}) (StoragePropertyConversion, error) {
+
+	sourceEndpoint := NewStorageConversionEndpoint(
+		sourceProperty.propertyType, string(sourceProperty.propertyName), knownLocals)
+	destinationEndpoint := NewStorageConversionEndpoint(
+		destinationProperty.propertyType, string(destinationProperty.propertyName), knownLocals)
+
+	conversion, err := createTypeConversion(sourceEndpoint, destinationEndpoint)
+
+	if err != nil {
+		return nil, errors.Wrapf(
+			err,
+			"trying to assign %s from %s",
+			destinationProperty.propertyName,
+			sourceProperty.propertyName)
+	}
+
+	return func(source dst.Expr, destination dst.Expr, ctx *CodeGenerationContext) []dst.Stmt {
+
+		var reader = astbuilder.Selector(source, string(sourceProperty.PropertyName()))
+		var writer = astbuilder.Selector(destination, string(destinationProperty.PropertyName()))
+
+		return conversion(reader, writer, ctx)
+	}, nil
+}
+
+// StorageTypeConversion generates the AST for a given conversion.
+// source is an expression to read the original value
```

Comment:
comments for parameter names don't match actual parameter names.

I do kinda prefer `source` + `destination` more than `reader`/`writer` but see how you got to `reader`/`writer` given the ubiquity of `source`/`destination`.